app/models/tag_filter.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+class TagFilter
+  attr_reader :params
+
+  def initialize(params)
+    @params = params
+  end
+
+  def results
+    scope = Tag.unscoped
+
+    params.each do |key, value|
+      next if key.to_s == 'page'
+
+      scope.merge!(scope_for(key, value.to_s.strip)) if value.present?
+    end
+
+    scope.order(id: :desc)
+  end
+
+  private
+
+  def scope_for(key, value)
+    case key.to_s
+    when 'context'
+      Tag.discoverable if value == 'directory'
+    when 'name'
+      Tag.matches_name(value)
+    when 'order'
+      scope_for_order(value)
+    when 'review'
+      scope_for_review(value)
```

Comment:
I'm not sure if these method names are okay...
I'm also not sure if this is the best way to call these methods.
Is there a better idea?

Reply:
This seems OK.